lib/rules/no-loop-func.js

```diff
@@ -188,6 +201,21 @@ module.exports = {
             const references = sourceCode.getScope(node).through;
             const unsafeRefs = references.filter(r => r.resolved && !isSafe(loopNode, r)).map(r => r.identifier.name);
```

Comment:
We can move this after `if (!(node.async || node.generator)) {}` as there's no need to find unsafe references if the function is going to be skipped.